lexicon/providers/powerdns.py

```diff
@@ -0,0 +1,204 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from .base import Provider as BaseProvider
+import requests
+import json
+import os
+
+# Lexicon PowerDNS Provider
+#
+# Author: Will Hughes, 2017
+#
+# API Docs: https://doc.powerdns.com/md/httpapi/api_spec/
+#
+# Implementation notes:
+# * The PowerDNS API does not assign a unique identifier to each record in the way
+#   that Lexicon expects. We work around this by creating an ID based on the record
+#   name, type and content, which when taken together are always unique
+# * The PowerDNS API has no notion of 'create a single record' or 'delete a single
+#   record'. All operations are either 'replace the RRSet with this new set of records'
+#   or 'delete all records for this name and type. Similarly, there is no notion of
+#   'change the content of this record', because records are identified by their name,
+#   type and content.
+# * The API is very picky about the format of values used when creating records:
+# ** CNAMEs must be fully qualified
+# ** TXT, LOC records must be quoted
+#   This is why the _clean_content and _unclean_content methods exist, to convert
+#   back and forth between the format PowerDNS expects, and the format Lexicon uses
+
+
+def ProviderParser(subparser):
+    subparser.add_argument("--auth-token", help="specify token used authenticate")
+    subparser.add_argument("--pdns-server", help="URI for PowerDNS server")
+    subparser.add_argument("--pdns-server-id", help="Server ID to interact with")
+
+
+class Provider(BaseProvider):
+
+    def __init__(self, options, provider_options={}):
+        super(Provider, self).__init__(options)
+
+        self.api_endpoint = options.get('pdns_server')
+        if self.api_endpoint is None:
+            self.api_endpoint = os.environ.get("LEXICON_POWERDNS_PDNS_SERVER")
```

Comment:
Lines 42 & 43 are unnecessary. `options.get('pdns_server')` on line 41 will be populated with the variable from environmental variable if not passed via the CLI. 

https://github.com/AnalogJ/lexicon/blob/master/lexicon/client.py#L50-L62